python-threatexchange/threatexchange/storage/local_dbm.py

```diff
@@ -0,0 +1,113 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+
+"""
+Simple dbm implementation of the storage interface.
+"""
+
+import dbm
+from enum import Enum
+from dataclasses import dataclass
+import typing as t
+from pathlib import Path
+
+from threatexchange.utils import dataclass_json
+from threatexchange.storage import interfaces as iface
+from threatexchange.signal_type.signal_base import SignalType
+from threatexchange.content_type.content_base import ContentType
+from threatexchange.exchanges.signal_exchange_api import TSignalExchangeAPICls
+from threatexchange.exchanges.impl.static_sample import StaticSampleSignalExchangeAPI
+from threatexchange.signal_type.pdq.signal import PdqSignal
+from threatexchange.signal_type.md5 import VideoMD5Signal
+from threatexchange.content_type.photo import PhotoContent
+from threatexchange.content_type.video import VideoContent
+
+
+class _DbType(Enum):
+    SIGNAL_TYPE = "signal_type"
+    CONTENT_TYPE = "content_type"
+    EXCHANGE_TYPE = "exchange_type"
+    EXCHANGES = "exchanges"
+    BANKS = "banks"
+    INDEX = "index"
+
+
+@dataclass
+class _SignalTypeCfg:
+    enabled_ratio: float = 1.0
+
+
+# TODO - eventually to unified store
+class DBMStore(iface.ISignalTypeConfigStore):
+    """
+    Local machine storage based on python dbm library,
+
+    A simple one-machine storage interface, which persists on disk.
+    Useful for testing or single machine work.
+    """
+
+    _folder: Path
+    signal_types: t.Mapping[str, t.Type[SignalType]]
+    content_types: t.Mapping[str, t.Type[ContentType]]
+    exchange_types: t.Mapping[str, TSignalExchangeAPICls]
+
+    def __init__(
+        self,
+        folder: Path,
+        *,
+        signal_types: t.Sequence[t.Type[SignalType]] | None = None,
+        content_types: t.Sequence[t.Type[ContentType]] | None = None,
+        exchange_types: t.Sequence[TSignalExchangeAPICls] | None = None,
```

Comment:
`|` not supported in old codez
```suggestion
        signal_types: t.Optional[t.Sequence[t.Type[SignalType]]] = None,
        content_types: t.Optional[t.Sequence[t.Type[ContentType]]] = None,
        exchange_types: t.Optional[t.Sequence[TSignalExchangeAPICls]] = None,
```